common/src/main/java/io/pravega/common/util/CompositeBufferView.java

```diff
@@ -185,6 +201,69 @@ public int readBytes(ByteArraySegment segment) {
             return 0;
         }
 
+        @Override
+        public byte readByte() throws EOFException {
+            BufferView.Reader current = getCurrent();
+            if (current == null) {
+                throw new EOFException();
+            }
+
+            byte result = current.readByte();
+            this.available--;
+            assert this.available >= 0;
+            return result;
+        }
+
+        @Override
+        public int readInt() throws EOFException {
+            BufferView.Reader current = getCurrent();
+            if (current != null && current.available() >= Integer.BYTES) {
+                this.available -= Integer.BYTES;
+                assert this.available >= 0;
```

Comment:
um, this assert it really obvious... (like just based on the if condition above)

Reply:
ok, removed.